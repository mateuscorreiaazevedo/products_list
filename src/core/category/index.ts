export * from './entities/category.entity'
export * from './contracts/category.contract'
export * from './services/http/http-client-category.service'
export * from './mappers/category.mapper'
export * from './dtos/list-categories.dto'
export * from './services/mock/mock-category.service'
export * from './usecases/list-categories.usecase'
export * from './factories/category.factory'
