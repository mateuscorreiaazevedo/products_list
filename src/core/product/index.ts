export * from './services/http/http-client-product.service'
export * from './usecases/list-products.usecase'
export * from './contracts/products.contract'
export * from './factories/product.factory'
export * from './entities/product.entity'
