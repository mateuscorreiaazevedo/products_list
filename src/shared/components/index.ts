export * from './ui/stack'
export * from './ui/grid'
export * from './ui/text'
export * from './ui/card'
export * from './ui/img'
export * from './ui/buttons'

export * from './layout'
export * from './map-list'
