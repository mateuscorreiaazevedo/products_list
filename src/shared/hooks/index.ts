export * from './cookie-storage'
export * from './debounce'
