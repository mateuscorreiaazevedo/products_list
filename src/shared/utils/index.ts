export * from './helpers/http-status-helper'
export * from './currency-builder'
export * from './string-builder'
