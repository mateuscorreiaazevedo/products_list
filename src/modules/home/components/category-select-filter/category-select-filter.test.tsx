import { CategoryFactory, type CategoryItemDTO } from '@/core/category'
import { renderWithTheme } from '@/shared/tests'
import { CategorySelectFilter } from '.'

const mockCategories = async () => await CategoryFactory.mock().listCategories()

const renderComponent = (data: CategoryItemDTO[]) => {
  return renderWithTheme(<CategorySelectFilter categories={data} />)
}

describe('CategorySelectFilter', () => {
  test('Should return if the component is being rendered', async () => {
    const categories = await mockCategories()
    const result = renderComponent(categories.data)

    expect(result).toMatchSnapshot()
    expect(result.getByTestId('select-value')).toBeTruthy()
  })
})
