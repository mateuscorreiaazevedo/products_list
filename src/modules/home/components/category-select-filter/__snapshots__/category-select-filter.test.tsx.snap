// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelectFilter Should return if the component is being rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gjcoXW kpPONA"
        gap="1.6rem"
      >
        <button
          aria-autocomplete="none"
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-label="category"
          class="sc-hRDKVd hrUef"
          data-placeholder=""
          data-state="closed"
          data-testid="select-trigger"
          dir="ltr"
          role="combobox"
          type="button"
        >
          <span
            data-testid="select-value"
            style="pointer-events: none;"
          >
            Selecionar categoria
          </span>
          <span
            aria-hidden="true"
          >
            <svg
              class="lucide lucide-chevron-down"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gjcoXW kpPONA"
      gap="1.6rem"
    >
      <button
        aria-autocomplete="none"
        aria-controls="radix-:r0:"
        aria-expanded="false"
        aria-label="category"
        class="sc-hRDKVd hrUef"
        data-placeholder=""
        data-state="closed"
        data-testid="select-trigger"
        dir="ltr"
        role="combobox"
        type="button"
      >
        <span
          data-testid="select-value"
          style="pointer-events: none;"
        >
          Selecionar categoria
        </span>
        <span
          aria-hidden="true"
        >
          <svg
            class="lucide lucide-chevron-down"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
